test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -288,3 +267,86 @@ func TestVLANNetwork(t *testing.T) {
 	}
 	testSecondaryNetwork(t, networkTypeVLAN, pods)
 }
+
+func (data *testData) assignIP(clientset *kubernetes.Clientset) error {
+	e2eTestData := data.e2eTestData
+	namespace := e2eTestData.GetTestNamespace()
+
+	for _, testPod := range data.pods {
+		_, err := e2eTestData.PodWaitFor(defaultTimeout, testPod.podName, namespace, func(pod *corev1.Pod) (bool, error) {
+			if pod.Status.Phase != corev1.PodRunning {
+				return false, nil
+			}
+			podIPs, err := data.listPodIPs(testPod)
+			if err != nil {
+				return false, err
+			}
+			ip, exists := podIPs["eth1"]
+			if !exists || ip == nil {
+				logs.Infof("IP not available for interface 'eth1' in Pod %s, retrying...", testPod.podName)
+				return false, nil
+			}
+
+			node, err := clientset.CoreV1().Nodes().Get(context.TODO(), testPod.nodeName, metav1.GetOptions{})
+			if err != nil {
+				return false, fmt.Errorf("error when getting cluster Node %s: %v", testPod.nodeName, err)
+			}
+			eni, has := node.Labels["eni-id"]
+			if !has {
+				return false, fmt.Errorf("the label `eni-id` not found in cluster Node: %s", testPod.nodeName)
+			}
+
+			if err := aws.AssignIPToEC2ENI(eni, ip.String()); err != nil {
+				return false, err
+			}
+			logs.Infof("assigned private ip addresses %s to interface %s", ip, eni)
+			return true, nil
+		})
+		if err != nil {
+			return fmt.Errorf("error when waiting for secondary IPs for Pod %+v: %v", testPod, err)
```

Comment:
use the `%w` formatter when wrapping errors

Reply:
Done.